README.md

```diff
@@ -26,7 +33,60 @@ struct Person {
 }
 
 fn main() {
-    let conn = Connection::connect("postgres://postgres@localhost", SslMode::None).unwrap();
+    let uri = "postgres://postgres@localhost";
+    let conn = match Connection::connect(uri, SslMode::None) {
+        Ok(conn) => conn,
+        Err(e) => {
+            println!("Connection error: {}", e);
+            return;
+        }
+    };
+    conn.execute("CREATE TABLE person (
+                    id              SERIAL PRIMARY KEY,
+                    name            VARCHAR NOT NULL,
+                    data            BYTEA
+                  )", &[]).unwrap();
+    let me = Person {
+        id: 0,
+        name: "Steven".to_string(),
+        data: None,
+    };
+    conn.execute("INSERT INTO person (name, data) VALUES ($1, $2)",
+                 &[&me.name, &me.data]).unwrap();
+    for row in &conn.query("SELECT id, name, data FROM person", &[]).unwrap() {
+        let person = Person {
+            id: row.get(0),
+            name: row.get(1),
+            data: row.get(2),
+        };
+        println!("Found person {}", person.name);
+    }
+}
+```
+
+Same example as above but with SSL
+```rust
+extern crate postgres;
+
+use postgres::{Connection, SslMode};
+use openssl::ssl::{SslContext, SslMethod};
+
+struct Person {
+    id: i32,
+    name: String,
+    data: Option<Vec<u8>>,
+}
+
+fn main() {
+    let context = SslContext::new(SslMethod::Sslv23).unwrap();
```

Comment:
Unless I misunderstand I feel that's a separate issue as it's a bug in the
openssl crate and not the postgres crate

On Aug 14, 2016 4:20 PM, "Steven Fackler" <notifications@github.com> wrote:

> In README.md
> <https://github.com/sfackler/rust-postgres/pull/194#discussion_r74710701>:
>
> > +
> > +Same example as above but with SSL
> > +```rust
> > +extern crate postgres;
> > +
> > +use postgres::{Connection, SslMode};
> > +use openssl::ssl::{SslContext, SslMethod};
> > +
> > +struct Person {
> > +    id: i32,
> > +    name: String,
> > +    data: Option<Vec<u8>>,
> > +}
> > +
> > +fn main() {
> > +    let context = SslContext::new(SslMethod::Sslv23).unwrap();
>
> I'm a bit hesitant to use this specifically as an example since it doesn't
> do any certificate validation at all. OpenSSL unfortunately gives you all
> of the rope you need to hang yourself :(. The situation here should be
> better in the next release with the new logic in here:
> https://github.com/sfackler/rust-postgres/blob/breaks/src/
> io/openssl.rs#L30-L39
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sfackler/rust-postgres/pull/194/files/d5bfc161a799f5044c6dd93df8a394d43ffcb828#r74710701>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABMFFmaSulKT6-tNq2zEtyHK7O1BA0loks5qf6KtgaJpZM4JeYP7>
> .
>